UM/Settings/Interfaces.py

```diff
@@ -1,11 +1,40 @@
 # Copyright (c) 2016 Ultimaker B.V.
 # Uranium is released under the terms of the AGPLv3 or higher.
 
+from collections import deque
 from typing import List, Dict, Any, Optional
 
 import UM.Decorators
 from UM.Signal import Signal
 
+##  Context for evaluating a property value
+#   It contains:
+#     1. a stack of containers during the evaluation in the function call stack fashion
+#     2. a context dictionary which contains all the current context
+#
+class PropertyEvaluationContext:
+
+    def __init__(self, source_stack = None):
+        self._stack_of_containers = deque()
+        if source_stack is not None:
+            self._stack_of_containers.append(source_stack)
+        self._context = {}
+
+    @property
+    def root_stack(self):
```

Comment:
Code style: Function names should be lower camel case.

Reply:
Fixed in bf9acbc3a227aaf4be1ef9873ef0b69926e51468